src/app/components/media/CreateMediaSource.js

```diff
@@ -0,0 +1,493 @@
+  };
+
+  const handleCancelOrSave = () => {
+    onCancel();
+  };
+
+  const handleCancelDialog = () => {
+    setDialogOpen(false);
+    setValidatePrimaryLinkExist(false);
+  };
+
+  const handleSubmitDialog = () => {
+    relateToExistingSource({ dbid: existingSource.id });
+  };
+
+  const handleSave = () => {
+    if (!submitDisabled && validateLinks() && validatePrimaryLink()) {
+      setSubmitDisabled(true);
+      const urls = [];
+      let newLinks = [primaryUrl].concat(links);
+      newLinks = newLinks.filter(link => !!link.url.trim());
+      newLinks.forEach((link) => {
+        urls.push(link.url);
+      });
+
+      const onFailure = (transaction) => {
+        setSubmitDisabled(false);
+        const error = getErrorObjects(transaction);
+        if (Array.isArray(error) && error.length > 0) {
+          if (error[0].code === CheckError.codes.DUPLICATED) {
+            setDialogOpen(true);
+            setExistingSource(error[0].data);
+          } else {
+            const messageError = getErrorMessage(transaction, <GenericUnknownErrorMessage />);
+            setMessage(messageError);
+          }
+        }
+      };
+
+      const onSuccess = () => {
+        handleCancelOrSave();
+      };
+
+      Relay.Store.commitUpdate(
+        new CreateSourceMutation({
+          name: sourceName,
+          slogan: sourceName,
+          image,
+          urls,
+          validate_primary_link_exist,
+          project_media: media,
+        }),
+        { onSuccess, onFailure },
+      );
+    }
+    return true;
+  };
+
+
+  return (
+    <div id="media-source-create-new">
+      <div className={classes.CancelSaveRow}>
+        <Button
+          className="source__edit-cancel-button"
+          onClick={handleCancelOrSave}
+        >
+          <FormattedMessage {...globalStrings.cancel} />
+        </Button>
+        <Button
+          variant="contained"
+          color="primary"
+          className="source__edit-save-button"
+          onClick={handleSave}
+          disabled={submitDisabled}
+        >
+          <FormattedMessage {...globalStrings.save} />
+        </Button>
+      </div>
+      <Message message={message} />
+      <div className={classes.headerRow}>
+        <StyledTwoColumns>
+          <StyledSmallColumn>
+            <SourcePicture
+              type="user"
+              className="source__avatar"
+            />
+            {!editProfileImg ?
+              <StyledAvatarEditButton className="source__edit-avatar-button">
+                <Button
+                  onClick={() => { setEditProfileImg(true); }}
+                  color="primary"
+                >
+                  <FormattedMessage {...globalStrings.edit} />
+                </Button>
+              </StyledAvatarEditButton>
+              : null}
+          </StyledSmallColumn>
+          <StyledBigColumn>
+            <div className="source__primary-info">
+              { editProfileImg ?
+                <div>
```

Comment:
Let's use `<React.Fragment>` instead of `<div>` when the sole purpose is to group tags.
See why at https://stackoverflow.com/a/47762117/4642548

Reply:
Done